geode-core/src/main/java/org/apache/geode/management/internal/configuration/functions/GetClusterConfigurationFunction.java

```diff
@@ -27,6 +27,7 @@
 
 public class GetClusterConfigurationFunction implements InternalFunction {
   private static final Logger logger = LogService.getLogger();
+  private static final long serialVersionUID = 6332908511113951823L;
```

Comment:
Is there any way to fix this that doesn't involve Java serialization at all?

Reply:
I think that implementing `getId()` on the `GetClusterConfigurationFunction` class and registering the function using `FunctionService.registerFunction()` on each member should allow it to not use Java serialization. I don't know how feasible that would be in terms of preserving backwards compatibility though.

Reply:
@DonalEvans it will be tough, I think this needs to be solved as a larger scale. As you can see that the Serializable comes  from InternalFunction class and we have all ton of these internal functions that are needed for Geode operation. 
 @pivotal-jbarrett I went with the status quo with this solution, the function in the already released older versions were using Java Serialization hence I kept the solution in the same serialization domain. If there is was to use two different type of serialization for the class in different versions.

Reply:
Yes, this is a MUCH larger problem we will need to solve someday. Adding to my todo list...

Reply:
The javadoc on the `Function` interface says
>\* Even though this interface extends Serializable, functions will only be serialized if they are
>\* not registered. For best performance it is recommended that you implement {@link #getId()} to
>\* return a non-null identifier and register your function using {@link
>\* FunctionService#registerFunction(Function)} or the cache.xml <code>function</code> element.

This seems to imply that if the function is registered, the UUID shouldn't matter, because it's never used. The tricky part here would be that the function is not being registered on older versions, so the older version would need to be patched to have the function be registered there too for it to work. I think this would be the better long-term solution, but possibly not feasible right now.

At the least, it might be good to have the function be registered in this version, so that future versions can benefit from the improved performance of not having to use Java serialization.

Reply:
@DonalEvans that will work but we will still need to add the serialVersionUID in that case too so that the already released versions are able to operate during the rolling update.
That effort can be different PR where we add getID to all the missing IDs in the InternalFunction